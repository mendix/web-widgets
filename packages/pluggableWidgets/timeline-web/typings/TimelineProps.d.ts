/**
 * This file was generated from Timeline.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Framework Team
 */
import { ComponentType, CSSProperties } from "react";
import { DynamicValue, ListValue, ListActionValue, ListAttributeValue, ListExpressionValue, ListWidgetValue, WebIcon } from "mendix";

export type GroupByKeyEnum = "day" | "month" | "year";

export type GroupByDayOptionsEnum = "dayName" | "dayMonth" | "fullDate";

export type GroupByMonthOptionsEnum = "month" | "monthYear";

export type UngroupedEventsPositionEnum = "beginning" | "end";

export interface TimelineContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    data: ListValue;
    title?: ListExpressionValue<string>;
    description?: ListExpressionValue<string>;
    timeIndication?: ListExpressionValue<string>;
    customVisualization: boolean;
    icon?: DynamicValue<WebIcon>;
    groupEvents: boolean;
    groupAttribute?: ListAttributeValue<Date>;
    groupByKey: GroupByKeyEnum;
    groupByDayOptions: GroupByDayOptionsEnum;
    groupByMonthOptions: GroupByMonthOptionsEnum;
    ungroupedEventsPosition: UngroupedEventsPositionEnum;
    customIcon?: ListWidgetValue;
    customGroupHeader?: ListWidgetValue;
    customTitle?: ListWidgetValue;
    customEventDateTime?: ListWidgetValue;
    customDescription?: ListWidgetValue;
    onClick?: ListActionValue;
}

export interface TimelinePreviewProps {
    /**
     * @deprecated Deprecated since version 9.18.0. Please use class property instead.
     */
    className: string;
    class: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    data: {} | { type: string } | null;
    title: string;
    description: string;
    timeIndication: string;
    customVisualization: boolean;
    icon: { type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; } | null;
    groupEvents: boolean;
    groupAttribute: string;
    groupByKey: GroupByKeyEnum;
    groupByDayOptions: GroupByDayOptionsEnum;
    groupByMonthOptions: GroupByMonthOptionsEnum;
    ungroupedEventsPosition: UngroupedEventsPositionEnum;
    customIcon: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    customGroupHeader: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    customTitle: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    customEventDateTime: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    customDescription: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    onClick: {} | null;
}
