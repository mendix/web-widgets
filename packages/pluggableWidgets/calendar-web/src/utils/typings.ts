import { ObjectItem } from "mendix";
import { CalendarProps as ReactCalendarProps } from "react-big-calendar";
import { withDragAndDropProps } from "react-big-calendar/lib/addons/dragAndDrop";

export interface CalendarEvent {
    title: string;
    start: Date;
    end: Date;
    allDay: boolean;
    color?: string;
    item: ObjectItem;
}

export type EventDropOrResize = {
    event: CalendarEvent;
    start: Date;
    end: Date;
};

export interface DragAndDropCalendarProps<TEvent extends object = Event, TResource extends object = object>
    extends ReactCalendarProps<TEvent, TResource>,
        withDragAndDropProps<TEvent, TResource> {}
