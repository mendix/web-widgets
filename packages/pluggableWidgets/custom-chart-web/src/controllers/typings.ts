import { ActionValue, EditableValue } from "mendix";

export interface ControllerProps {
    configurationOptions: string;
    dataAttribute?: EditableValue<string>;
    dataStatic: string;
    eventDataAttribute?: EditableValue<string>;
    layoutAttribute?: EditableValue<string>;
    layoutStatic: string;
    onClick?: ActionValue;
    sampleData: string;
    sampleLayout: string;
}
