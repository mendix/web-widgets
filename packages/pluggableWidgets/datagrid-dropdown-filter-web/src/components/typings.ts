import { RefFilterStore } from "@mendix/widget-plugin-dropdown-filter/stores/RefFilterStore";
import { StaticSelectFilterStore } from "@mendix/widget-plugin-dropdown-filter/stores/StaticSelectFilterStore";

export interface EnumFilterAPI {
    filterStore: StaticSelectFilterStore;
    parentChannelName?: string;
}

export interface RefFilterAPI {
    filterStore: RefFilterStore;
    parentChannelName?: string;
}
