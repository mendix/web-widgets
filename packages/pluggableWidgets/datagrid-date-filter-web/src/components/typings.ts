import { Date_InputFilterInterface } from "@mendix/widget-plugin-filtering/typings/InputFilterInterface";

export interface DateFilterProps {
    filterStore: Date_InputFilterInterface;
    parentChannelName?: string;
}
