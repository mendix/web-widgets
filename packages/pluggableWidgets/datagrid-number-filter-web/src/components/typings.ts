import { ChangeEventHandler, CSSProperties } from "react";
import { FilterType } from "../../typings/FilterType";
import { Big } from "big.js";

interface FilterProps {
    adjustable: boolean;
    defaultFilterType: FilterType;
    className?: string;
    id?: string;
    placeholder?: string;
    screenReaderButtonCaption?: string;
    screenReaderInputCaption?: string;
    tabIndex?: number;
    styles?: CSSProperties;
}

export interface FilterInputProps extends FilterProps {
    onFilterTypeClick: (type: FilterType) => void;
    onInputChange: ChangeEventHandler<HTMLInputElement>;
    inputValue: string;
    inputRef?: React.ClassAttributes<HTMLInputElement>["ref"];
    inputDisabled?: boolean;
}

export interface FilterComponentProps extends FilterProps {
    inputChangeDelay: number;
    value?: Big;
    updateFilters?: (value: Big | undefined, type: FilterType) => void;
    name: string;
    datagridChannelName?: string;
}
