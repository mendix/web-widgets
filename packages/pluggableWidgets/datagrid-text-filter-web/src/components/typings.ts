import { String_InputFilterInterface } from "@mendix/widget-plugin-filtering/typings/InputFilterInterface";

export interface StringFilterProps {
    filterStore: String_InputFilterInterface;
    parentChannelName?: string;
}
