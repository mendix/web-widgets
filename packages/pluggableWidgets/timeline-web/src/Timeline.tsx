import { createElement, ReactElement } from "react";
import { TimelineContainerProps } from "../typings/TimelineProps";
import "./ui/Timeline.scss";
import TimelineComponent from "./components/TimelineComponent";
import { withBasicItems } from "./hocs/withBasicItems";
import { withCustomItems } from "./hocs/withCustomItems";

const BasicView = withBasicItems(TimelineComponent);
const CustomView = withCustomItems(TimelineComponent);

export default function Timeline(props: TimelineContainerProps): ReactElement {
    if (props.customVisualization) {
        return <CustomView {...props} />;
    }

    return <BasicView {...props} />;
}
