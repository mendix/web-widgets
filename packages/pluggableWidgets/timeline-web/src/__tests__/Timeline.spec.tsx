import { shallow } from "enzyme";
import { createElement } from "react";
import { TimelineContainerProps } from "../../typings/TimelineProps";
import { buildListExpression, buildWidgetValue, ListValueBuilder } from "@mendix/widget-plugin-test-utils";
import Timeline from "../Timeline";

describe("Timeline", () => {
    const listValueBuilder = ListValueBuilder();
    const defaultProps: TimelineContainerProps = {
        name: "timeline",
        class: "test",
        tabIndex: 0,
        customVisualization: false,
        groupByKey: "day",
        data: listValueBuilder.simple(),
        groupEvents: true,
        title: buildListExpression("title1"),
        description: buildListExpression("description"),
        timeIndication: buildListExpression("time"),
        customIcon: buildWidgetValue(<img src={"test"} />),
        customGroupHeader: buildWidgetValue(<p>Custom Divider</p>),
        customTitle: buildWidgetValue(<p>Custom Title</p>),
        customEventDateTime: buildWidgetValue(<p>Custom Event Date Time</p>),
        customDescription: buildWidgetValue(<p>Custom Description</p>),
        groupByMonthOptions: "month",
        groupByDayOptions: "dayName",
        ungroupedEventsPosition: "end"
    };
    it("renders correctly with basic props", () => {
        const component = shallow(<Timeline {...defaultProps} />);
        expect(component).toMatchSnapshot();
    });
});
