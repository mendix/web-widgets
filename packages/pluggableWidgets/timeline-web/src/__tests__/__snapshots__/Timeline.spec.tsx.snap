// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline renders correctly with basic props 1`] = `
<DocumentFragment>
  <div
    class="widget-timeline-wrapper test"
  >
    <div
      class="timeline-date"
    >
      <div
        class="widget-timeline-events-wrapper widget-timeline-no-divider"
      >
        <ul>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <div
                class="widget-timeline-icon-circle"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  title1
                </p>
                <p
                  class="widget-timeline-description"
                >
                  description
                </p>
              </div>
            </div>
          </li>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <div
                class="widget-timeline-icon-circle"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  title1
                </p>
                <p
                  class="widget-timeline-description"
                >
                  description
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
